### DISCUSSION TITLE
Upgrade in Docker failed, 4.13 to 5.09

### DISCUSSION THREAD

**Discussion by ThomasKna** [globaleaks/globaleaks-whistleblowing-software#4196] [General]
latest Dockerfile, same volumes, build without errors, nothing else changed

globaleaks    | 2024-09-20T12:49:27+0000 [stdout#info] [E] ERROR: Cannot start GlobaLeaks. Please manually examine the exception.
globaleaks    | 2024-09-20T12:49:27+0000 [stdout#info] [E] EXCEPTION: (sqlite3.OperationalError) unable to open database file\n(Background on this error at: https://sqlalche.me/e/14/e3q8)

any idea whats excactly the problem is? volume is present, database is present.
-rw------- 1 1337 root 1000K Sep 19 11:17 globaleaks.db

still have a 4.13 Backup. possible to start the 5.09 from the scratch with this backup?

thanks, regards, thomas

**Comment by evilaliv3** [comment]
Thank you for your question.

I'm not sure which could be the problem, maybe for some reason you tempered the permissions?

For sure, any new image of globaleaks can be run with old data since we always include migration scripts that would update the data for you.

**Comment by ThomasKna** [comment]
I restarted version 5.10 from scratch using the backup from version 4.13. The database upgrade script progressed smoothly until version 68, at which point the following error occurred:

```
globaleaks    | 2024-09-23T15:09:56+0000 [stdout#info] [E] Failure: [Errno 1] Operation not permitted
globaleaks    | 2024-09-23T15:09:56+0000 [stdout#info] [E] Verbose exception traceback:
globaleaks    | 2024-09-23T15:09:56+0000 [stdout#info] [I] Traceback (most recent call last):\n\n  File "/usr/lib/python3.11/shutil.py", line 825, in move\n    os.rename(src, real_dst)\n\nPermissionError: [Errno 13] Permission denied: '/var/globaleaks/globaleaks/tmp/tmp/new.db' -> '/var/globaleaks/globaleaks/globaleaks.db'\n\n\nDuring handling of the above exception, another exception occurred:\n\n\nTraceback (most recent call last):\n\n  File "/usr/lib/python3/dist-packages/globaleaks/db/__init__.py", line 94, in update_db\n    migration.perform_migration(db_version)\n\n  File "/usr/lib/python3/dist-packages/globaleaks/db/migration.py", line 266, in perform_migration\n    shutil.move(new_db_file, db_file)\n\n  File "/usr/lib/python3.11/shutil.py", line 845, in move\n    copy_function(src, real_dst)\n\n  File "/usr/lib/python3.11/shutil.py", line 437, in copy2\n    copystat(src, dst, follow_symlinks=follow_symlinks)\n\n  File "/usr/lib/python3.11/shutil.py", line 376, in copystat\n    lookup("utime")(dst, ns=(st.st_atime_ns, st.st_mtime_ns),\n\nPermissionError: [Errno 1] Operation not permitted\n
globaleaks    | 2024-09-23T15:09:56+0000 [twisted.scripts._twistd_unix.UnixAppLogger#info] Server Shut Down.
```

 This seems to be related to permission issues, likely due to SELinux restrictions?

My Solution: I moved the restored backup into a subfolder, e.g., /var/globaleaks/data, and updated the working path accordingly in the Dockerfile.
